crates/ruff_python_formatter/src/other/comprehension.rs

```diff
@@ -1,12 +1,115 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
+use crate::comments::{leading_comments, trailing_comments, SourceComment};
+use crate::prelude::*;
+use crate::AsFormat;
+use crate::{FormatNodeRule, PyFormatter};
 use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::Comprehension;
+use rustpython_parser::ast::{Comprehension, Ranged};
 
 #[derive(Default)]
 pub struct FormatComprehension;
 
 impl FormatNodeRule<Comprehension> for FormatComprehension {
     fn fmt_fields(&self, item: &Comprehension, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let Comprehension {
+            range: _,
+            target,
+            iter,
+            ifs,
+            is_async,
+        } = item;
+
+        let comments = f.context().comments().clone();
+        let leading_item_comments = comments.leading_comments(item);
+
+        // TODO: why is this needed?
+        if let Some(leading_item_comment) = leading_item_comments.first() {
+            if leading_item_comment.line_position().is_own_line() {
+                hard_line_break().fmt(f)?;
+            }
+        }
+        leading_comments(leading_item_comments).fmt(f)?;
+
+        if *is_async {
+            write!(f, [text("async"), space()])?;
+        }
+
+        let dangling_item_comments = comments.dangling_comments(item);
+
+        let (before_target_comments, before_in_comments) = dangling_item_comments.split_at(
+            dangling_item_comments
+                .partition_point(|comment| comment.slice().end() < target.range().start()),
+        );
+
+        let trailing_in_comments = comments.dangling_comments(iter);
+        write!(
+            f,
+            [
+                text("for"),
+                trailing_comments(before_target_comments),
+                group(&self::format_args!(
+                    soft_line_break_or_space(),
+                    &target.format(),
```

Comment:
```suggestion
                    target.format(),
```

Reply:
👍 clippy usually complains about these (but not this one)